backend/tournesol/lib/public_dataset.py

```diff
@@ -69,6 +70,10 @@ def get_comparisons_data(poll_name: str) -> QuerySet:
           -- keep only public ratings
           AND rating_1.is_public = true
           AND rating_2.is_public = true
+          -- excluse current week comparisons
+          AND datetime_add < DATE_TRUNC('week', now())
```

Comment:
This query looks good to me.

Let's just keep in mind that the dataset is currently scheduled to be generated on Sundays at 23:55 UTC. 
https://github.com/tournesol-app/tournesol/blob/56f19a59bfcecd54865c5d3d060fa3efd846c6e8/infra/ansible/inventory.yml#L193

I think that @GresilleSiffle's intention was to not include any comparison from the new week, while avoiding a conflict with the backup script running on 00:00. Now that the SQL query excludes comparisons submitted in the current week, it should be okay to generate the dataset on Mondays, let's say at 00:02.

(we can address the change in inventory.yaml separately)

Reply:
makes sense, changed the yaml in my latest commit https://github.com/tournesol-app/tournesol/pull/1324/commits/16097207e2ad5054f858c12ee5f5ecbffc55c7c4 let me know if this is how you want to proceed

Reply:
> I think that @GresilleSiffle's intention was to not include any comparison from the new week, while avoiding a conflict with the backup script running on 00:00. Now that the SQL query excludes comparisons submitted in the current week, it should be okay to generate the dataset on Mondays, let's say at 00:02.

Exactly, scheduling the dataset creation before Monday was a way to workaround the absence of filter by date in the public dataset lib.

I thought about improving the lib to make the functions able to receive the `until` date as a parameter, to make it more flexible, but excluding the comparison directly in SQL is also good for me.